src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -727,18 +729,14 @@ def generate_ssh_keys_ed25519(private_key_filepath, public_key_filepath):
         encryption_algorithm=serialization.NoEncryption()
     )
 
-    with os.fdopen(os.open(private_key_filepath, flags=os.O_WRONLY | os.O_TRUNC | os.O_CREAT, mode=384, ), "w", ) as f:
-        f.write(
-            private_bytes.decode()
-        )
-    os.chmod(private_key_filepath, 0o600)
-
-    with open(public_key_filepath, 'w') as public_key_file:
-        s = public_key.public_bytes(
-            encoding=serialization.Encoding.OpenSSH,
-            format=serialization.PublicFormat.OpenSSH)
-        public_key = s.decode(encoding="utf8").replace("\n", "")
-        public_key_file.write(public_key)
-    os.chmod(public_key_filepath, 0o644)
+    with os.fdopen(_open(private_key_filepath, 0o600), "w") as f:
+        f.write(private_bytes.decode())
+
+    s = public_key.public_bytes(
+        encoding=serialization.Encoding.OpenSSH,
+        format=serialization.PublicFormat.OpenSSH)
+    public_key = s.decode(encoding="utf8").replace("\n", "")
```

Comment:
`public_bytes` contains `bytes` which can be directly written to a file opened as `wb`, eliminating the necessity to decode it to a `str`.